coalib/parsing/filters/CanDetectFilter.py

```diff
@@ -0,0 +1,21 @@
+from coalib.parsing.filters.FilterBase import FilterBase
+
+
+class CanDetectFilter(FilterBase):
+
+    def filter_section_bears(self, bears, args):
+        """
+        Filters the bears by can_detect.
+
+        :param bears:       The dictionary of the sections as keys and list of
+                            bears as values.
+        :param args:        detects that bears are being filtered on.
+        :return:            New dictionary with filtered out bears that matches
+                            any detect from detects.
+        """
+        detects = {detect.lower() for detect in args}
+        return {section:
+                tuple(bear for bear in bears[section]
+                      if {detect.lower()
+                          for detect in bear.CAN_DETECT} & detects)
+                for section in bears.keys()}
```

Comment:
Btw: -->
```python
{section:
 tuple(bear for bear in bs
       if {detect.lower() for detect in bear.CAN_DETECT} & detects)
 for bs in bears.values()}
```
1. Iterating over `bears.keys()` is the same as iterating over just `bears`.
2. You just use the values, so just use `bears.values()` :)